### PR TITLE
Add Gulp build system, and move XPI source files into src/ dir

### PR DESCRIPTION
**Pull request by jaredhirsch**
Replaces #52 by rebasing. @pdehaan will still get his name on the commit, it was an easy rebase :-)

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const del = require('del');
+const eslint = require('gulp-eslint');
+const gulp = require('gulp');
+const zip = require('gulp-zip');
+
+gulp.task('build', ['lint', 'clean:dist'], function () {
+  return gulp.src('src/*')
```
Comment (jaredhirsch) [inline gulpfile.js]: @pdehaan: hmm, building the XPI doesn't recurse into subdirectories. I think this line needs to be

```
gulp.src('src/**/*')
```

**Comment by jaredhirsch** [comment]
Looks like the new files from the rebased commit have to get moved over, too (lib/Broker and lib/Transport)

File: gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const del = require('del');
+const eslint = require('gulp-eslint');
+const gulp = require('gulp');
+const zip = require('gulp-zip');
+
+gulp.task('build', ['lint', 'clean:dist'], function () {
+  return gulp.src('src/**')
```
Comment (6a68) [inline gulpfile.js]: @pdehaan hmm, still seems busted. But `src/**/*` seems to work for me. Give it a try, let me know if you see the same behavior.

**Comment by jaredhirsch** [comment]
@pdehaan thanks for rebasing! I thought I saw a bug, but it looks like I have the wrong version of the commit. Trying again...

**Comment by pdehaan** [comment]
1__pdehaan_pdehaan____dev_tmp_del_univ-addon-gulp__zsh__and__universal-search-addon__add_gulp_build_system__and_move_xpi_source_files_into_src__dir___55__-_peterdehaan_gmail_com_-_gmail_and_dropbox_-_screenshot_2015-07-21_14_27_24_png

**Comment by pdehaan** [comment]
@6a68 I'm happy to change the glob, but I'm seeing `chrome/content/*`, and `chrome/skin/*`, and `lib/ui/*` so it looks like it's including sub-sub-dirs for me locally.

**Comment by jaredhirsch** [comment]
@pdehaan yeah, I needed to fetch & force update my branch. This looks awesome, thanks!

Merging like

**Pull request merged**